src/main/java/com/scrive/pdftools/tools/flattener/LambdaFlattener.java

```diff
@@ -0,0 +1,40 @@
+package com.scrive.pdftools.tools.flattener;
+
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.exceptions.BadPasswordException;
+import com.itextpdf.text.pdf.PdfReader;
+import com.itextpdf.text.pdf.PdfStamper;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class LambdaFlattener {
+
+    public static ByteArrayOutputStream execute(FlattenerSpec flatSpec) throws IOException {
+
+        final ByteArrayInputStream file = new ByteArrayInputStream(flatSpec.mainFileInput.getContent());
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        try {
+            PdfStamper stamper;
+            PdfReader reader = new PdfReader(file);
+            try {
+                stamper = new PdfStamper(reader, bos);
```

Comment:
also, I'm not familiar with `PdfStamper` but it seems it could also benefit from the try-with-resources treatment
```java
try (var stamper = new PdfStamper(reader, bos)) {
    ...
}
```

Reply:
The PdfStamper is not AutoCloseable so I don't think it can be used in a try-with-resources statement.

Reply:
PdfStamper can't be put in a try-with-resources because it's not AutoClosable

Reply:
In that case, let's borrow this pattern :) https://github.com/obiba/mica2/blob/c62918d6638feece02033887e6582ef4e259dc90/mica-core/src/main/java/org/obiba/mica/PdfUtils.java#L98

Reply:
@GustafNi ^^

Reply:
Good idea :). This looks better right?
 I am a little unsure about the catch and then use a new try-with-resources. Is there another way to adress this?